src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java

```diff
@@ -51,7 +52,12 @@ public KNNVectorFieldType(String name, Map<String, String> metadata, VectorDataT
 
     @Override
     public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup searchLookup, String format) {
-        throw new UnsupportedOperationException("KNN Vector do not support fields search");
```

Comment:
What does this format variable do? Do we have to add any check for that?

Reply:
Also do you see problem with backward compatibility?  Does this work for index created before 2.19?

Reply:
> What does this format variable do? Do we have to add any check for that?

I'm not sure exactly what it does, in my testing it was always null when `valueFetcher` was called for KNNVectorFieldType.

> Also do you see problem with backward compatibility? Does this work for index created before 2.19?

I don't see any reason why it would have issues with backward compatibility, but how would I test this to confirm it?